raft-tech/TANF-app
# Alertmanager Cloud.gov

## Summary of Changes
- Updated alertmanager manifest to pull alertmanager binary and start it in monolith mode
- Updated `plg/deploy.sh` to deploy alertmanager and handle networking
- Fixed some prometheus alert config errors
- Updated alertmanager config to notify receivers when an alert/group of alerts has been resolved
Pull request closes #3254

## How to Test
1. Deploy this PR to a dev environment. Note for testing purposes alertmanager is deployed in a dev environment. 
2. Log in with the CF CLI and run `cf add-network-policy alertmanager tdp-frontend-<ENV> --protocol tcp --port 80` and `cf add-network-policy tdp-frontend-<ENV> alertmanager --protocol tcp --port 8080` where `<ENV>` should be replaced with `raft`, or `qasp`, etc...
3. Log into the frontend as an OFA System Admin or Developer
4. Verify the `Alerts` tab is visible and proxies the alertmanager UI when clicked
5. See screenshots below for alerts that fired from the raft environment and the associated emails that were received

![Screenshot 2024-11-18 at 9 28 22 AM](https://github.com/user-attachments/assets/f7d9ddc5-c9fa-4256-8136-ba9a046abeef)
![Screenshot 2024-11-18 at 10 01 48 AM](https://github.com/user-attachments/assets/32e016c1-d248-47be-8920-a532ed7e2e37)

![Screenshot 2024-11-19 at 2 05 14 PM](https://github.com/user-attachments/assets/2741cf80-e1ed-43ab-98b0-9689cd950c17)
![Screenshot 2024-11-19 at 2 05 19 PM](https://github.com/user-attachments/assets/4a888f16-abb4-4d2d-afe2-c03d4711a2d4)


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] AM deployed in the production environment
- [x] AM UI accessible from any environment (dev, staging, and production)
- [x] AM alerts fire and route correctly from deployed environment

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
